rootex/framework/entity.h

```diff
@@ -18,8 +19,9 @@ class Entity
 	String m_Name;
 	HashMap<ComponentID, Ref<Component>> m_Components;
 	bool m_IsEditorOnly;
+	Ref<Script> m_Script;
```

Comment:
Check the validity of this pointer to see if a script is attached instead of the `hasScript` boolean inside `Script`